core/Testo.ml

```diff
@@ -194,6 +194,44 @@ let read_file = Helpers.read_file
 let with_temp_file = Temp_file.with_temp_file
 let with_capture = Store.with_capture
 
+(**************************************************************************)
+(* Hygiene *)
+(**************************************************************************)
+
+let getenv_string k =
+  match Sys.getenv_opt k with
+  | None -> ""
+  | Some v -> v
+
+let with_environment_variables env func =
+  let envs =
+    Helpers.list_map
+      (fun (k, tmp) ->
+        (* OCaml doesn't provide an 'unsetenv' function, so we won't
```

Comment:
Not sure how this will work then with SEMGREP_APP_TOKEN because
we might have it unset, but it will still be in the environment with an empty value.

Reply:
The only solutions I could think of are:
* implement our own `unsetenv` via C;
* make the user use a library that provides a layer on top of `Unix`.

Both are hard and/or impractical.

I don't see a great generic workaround other than having the application treat empty and unset environment variables identically.

Reply:
yes, at least our own code should not rely on just being an env var set; we must also look at the content